### PR TITLE
fix: issue with `user-valid` for form-components

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

Add correct `:user-valid` for checkbox and select

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/styles/_form-components.scss [hunk 1]
```diff
@@ -367,13 +360,13 @@ $floating-label-size: calc(
 	}
 }
 
-%check-element {
+@mixin set-default-check-element($selector) {
```
Comment (annsch) [inline packages/components/src/styles/_form-components.scss:363]: shouldn't this mixin also get used by `select` element?

File: packages/components/src/styles/_form-components.scss [hunk 2]
```diff
@@ -136,33 +136,24 @@ $floating-label-size: calc(
...
-	// We need to do a more complicated check for non-typed-input-elements
 	@if ($selector == input) {
-		$validations: ':not([type="text"], [type="password"], [type="search"]),:is([type="text"], [type="password"], [type="search"]):required';
-	}
-
-	// Check elements (radio and checkbox) should get selected by their native HTML tag `input`
-	@if ($selector == check) {
-		$selector: input;
+		// https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation
+		$validations: ":required, [minlength], [maxlength], [pattern]";
```
Comment (mfranzke) [inline packages/components/src/styles/_form-components.scss:148]:
```suggestion
		$validations: $validations + ", [minlength], [maxlength], [pattern]";
```

File: packages/components/src/styles/_form-components.scss [hunk 3]
```diff
@@ -136,33 +136,24 @@ $floating-label-size: calc(
...
+		@each $type in $input-valid-types {
+			$validations: $validations + ", [type=#{$type}]";
+		}
+	} @else if($selector == textarea) {
+		$validations: ":required, [minlength], [maxlength]";
```
Comment (mfranzke) [inline packages/components/src/styles/_form-components.scss:153]:
```suggestion
		$validations: $validations + ", [minlength], [maxlength]";
```

**Review by mfranzke — APPROVED** [review]

File: packages/components/src/styles/_form-components.scss (hunk 1 above)
Comment (nmerget) [reply packages/components/src/styles/_form-components.scss:363]: There is another mixin for input, textarea and select to handle the messages as well

**Review by mfranzke — APPROVED** [review]

**Pull request merged**